### PR TITLE
Fix Socket_setnonblocking() for Windows (At least if build with MSVC …

### PR DESCRIPTION
**Pull request by JuergenKosel**
…2010)

Partially revert commit a16cff63cc2c595bb9fd0334495953f5d0be7796 for file
src/Socket.c

- Do not include errno.h for Windows build
- Use errno as a local variable in Socket_setnonblocking() for Windows

Signed-off-by: Juergen Kosel <juergen.kosel@softing.com>

### CODE REVIEW

**Comment by JuergenKosel** [comment]
For the mqttas library I had an issue with visual studio 2010 builds:
Each connect attempt failed!
Running git bisect shows, that this problem arise by commit a16cff63cc2c595bb9fd0334495953f5d0be7796
This pull request fixes this issue.

**Comment by icraggs** [comment]
errno.h is included in Socket.h, in a section not applying to Windows, so that should not be needed anyway

**Comment by icraggs** [comment]
In fact, I'm guessing that just removing the inclusion of errno.h from Socket.c could fix the problem

File: src/Socket.c [hunk 1]
```diff
@@ -36,7 +36,9 @@
 #include "SSLSocket.h"
 #endif
 
+#if !defined(WIN32) && !defined(WIN64)
 #include <errno.h>
```
Comment (JuergenKosel) [inline src/Socket.c]: Should this be completely removed?

File: src/Socket.c [hunk 2]
```diff
@@ -99,6 +101,10 @@ int Socket_setnonblocking(int sock)
  */
 int Socket_error(char* aString, int sock)
 {
+#if defined(WIN32) || defined(WIN64)
+	int errno;
```
Comment (JuergenKosel) [inline src/Socket.c:102]: This definition of a local variable `errno` is also needed to solve the problem!

File: src/Socket.c (hunk 1 above)
Comment (icraggs) [reply src/Socket.c]: yes

File: src/Socket.c (hunk 2 above)
Comment (icraggs) [reply src/Socket.c:102]: but this is needed, yes.  It was removed by the VxWorks commit

**Comment by icraggs** [comment]
Do these changes now work for you on VS 2010?

**Comment by JuergenKosel** [comment]
Yes, with this changes mqtt works again for my MSVC 2010 build.

**Comment by icraggs** [comment]
Great!  Thanks.  I'll merge them in.

**Comment by dontsovcmc** [comment]
It works with MSVC 2013, thanks @JuergenKosel  :)

**Pull request merged**